manifests/params.pp

```diff
@@ -32,29 +32,58 @@
       }
     }
     'RedHat': {
-      $dnsdir             = '/etc'
-      $vardir             = '/var/named'
-      $optionspath        = '/etc/named/options.conf'
-      $zonefilepath       = "${vardir}/dynamic"
-      $localzonepath      = "${dnsdir}/named.rfc1912.zones"
-      $defaultzonepath    = 'unmanaged'
-      $publicviewpath     = "${dnsdir}/named/zones.conf"
-      $viewconfigpath     = "${dnsdir}/named/views"
-      $dns_server_package = 'bind'
-      $namedservicename   = 'named'
-      $user               = 'named'
-      $group              = 'named'
-      $rndcconfgen        = '/usr/sbin/rndc-confgen'
-      $named_checkconf    = '/usr/sbin/named-checkconf'
-      $sysconfig_file     = '/etc/sysconfig/named'
+      $user = 'named'
+      $group = 'named'
+
       $sysconfig_template = "dns/sysconfig.${facts['os']['family']}.erb"
       $sysconfig_startup_options = undef
       $sysconfig_disable_zone_checking = undef
 
       # This option is not relevant for RedHat
       $sysconfig_resolvconf_integration = undef
 
-      $dnssec_enable = if versioncmp($facts['os']['release']['major'], '9') >= 0 { undef } else { 'yes' }
+      if $dns::globals::scl {
+        $sclprovider = $dns::globals::scl.split('-')[0]
+        $sclenvname = $dns::globals::scl
```

Comment:
Perhaps skip this variable now and only use `$scl`? Since `params` inherits from `globals` you can just use `$scl`, though I've seen that people have difficulty with that so I wouldn't mind being explicit in it.